internal/handlers/staking_restrictions_hooks.go

```diff
@@ -0,0 +1,141 @@
+package handlers
+
+import (
+	"cosmossdk.io/math"
+	"github.com/cosmos/cosmos-sdk/simapp/helpers"
+	sdktypes "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	stakingkeeper "github.com/cosmos/cosmos-sdk/x/staking/keeper"
+	stakingtypes "github.com/cosmos/cosmos-sdk/x/staking/types"
+)
+
+// Assert proper implementation of StakingRestrictions
+var _ stakingtypes.StakingHooks = StakingRestrictionHooks{}
+
+const (
+	// The default concentration of bonded tokens a validator is allowed as a multiple of equal shares
+	DefaultConcentration = 5.5 // Amounts to ~8% with 68 validators
+	// Maximum Allowed Cap for Bonded stake of any single validator
+	DefaultMaxCapPercent = 0.33
+	// Minimum Allowed Cap for Bonded stake of any single validator
+	DefaultMinCapPercent = 0.05
+)
+
+type RestrictionOptions struct {
+	MaxConcentrationMultiple float32
+	MaxBondedCapPercent      float32
+	MinBondedCapPercent      float32
+}
+
+// DefaultRestictionOptions are default constraints that prevent single point of failure on validators
+var DefaultRestictionOptions = &RestrictionOptions{
+	MaxConcentrationMultiple: DefaultConcentration,
+	MaxBondedCapPercent:      DefaultMaxCapPercent,
+	MinBondedCapPercent:      DefaultMinCapPercent,
+}
+
+// UnlimitedRestrictionOptions are used to remove restrictions for validator staking limits from delegations
+var UnlimitedRestrictionOptions = &RestrictionOptions{
+	MaxConcentrationMultiple: 1.0,
+	MaxBondedCapPercent:      1.0,
+	MinBondedCapPercent:      1.0,
+}
+
+// Hooks wrapper struct for slashing keeper
+type StakingRestrictionHooks struct {
+	k    *stakingkeeper.Keeper
+	opts RestrictionOptions
+}
+
+// NewStakingRestrictionHooks configures a hook that validates changes to delegation modifications and
+// prevents concentration of voting power beyond configured limits on active validators.
+func NewStakingRestrictionHooks(k *stakingkeeper.Keeper, opts RestrictionOptions) StakingRestrictionHooks {
+	return StakingRestrictionHooks{k, opts}
+}
+
+// Verifies that the delegation would not cause the validator's voting power to exceed our staking distribution limits
+func (h StakingRestrictionHooks) AfterDelegationModified(ctx sdktypes.Context, delAddr sdktypes.AccAddress, valAddr sdktypes.ValAddress) error {
+	valCount := len(h.k.GetLastValidators(ctx))
+
+	// do not bother with limits on networks this small (or under simulation).
+	if valCount < 4 || ctx.ChainID() == helpers.SimAppChainID {
+		return nil
+	}
+
+	// bond limit is allowed to have a multiple of even shares of network bonded stake.
+	maxValidatorPercent := h.opts.MaxConcentrationMultiple * (1.0 / float32(valCount))
+
+	// check the capped bond amount is within the overall range limits.
```

Comment:
The code actually `sets` the the amount within the range. 

Should say something like...
```suggestion
	// set the capped bond amount within the overall range limits.
```

Reply:
the comment on the conditional is for the checks.
```
	// check the capped bond amount is within the overall range limits.
	if maxValidatorPercent > h.opts.MaxBondedCapPercent {
```